lte/gateway/c/oai/tasks/amf/amf_config.c

```diff
@@ -141,7 +152,646 @@ void amf_config_init(amf_config_t* config) {
+        config_aA->m5g_nas_config.enable_apn_correction = parse_bool(astring);
+      }
+
+      // Parsing APN CORRECTION MAP
+      if (config_aA->m5g_nas_config.enable_apn_correction) {
+        subsetting = config_setting_get_member(
+            setting, MME_CONFIG_STRING_NAS_APN_CORRECTION_MAP_LIST);
+        config_aA->m5g_nas_config.m5g_apn_map_config.nb = 0;
+        if (subsetting != NULL) {
+          num = config_setting_length(subsetting);
+          OAILOG_INFO(
+              LOG_MME_APP, "Number of apn correction map configured =%d\n",
+              num);
+          AssertFatal(
+              num <= MAX_APN_CORRECTION_MAP_LIST,
+              "Number of apn correction map configured:%d exceeds the maximum "
+              "number supported"
+              ":%d \n",
+              num, MAX_APN_CORRECTION_MAP_LIST);
+
+          for (i = 0; i < num; i++) {
+            sub2setting = config_setting_get_elem(subsetting, i);
+            if (sub2setting != NULL) {
+              if ((config_setting_lookup_string(
+                      sub2setting,
+                      MME_CONFIG_STRING_NAS_APN_CORRECTION_MAP_IMSI_PREFIX,
+                      (const char**) &imsi_prefix))) {
+                if (config_aA->m5g_nas_config.m5g_apn_map_config.apn_map[i]
+                        .imsi_prefix) {
+                  bassigncstr(
+                      config_aA->m5g_nas_config.m5g_apn_map_config.apn_map[i]
+                          .imsi_prefix,
+                      imsi_prefix);
+                } else {
+                  config_aA->m5g_nas_config.m5g_apn_map_config.apn_map[i]
+                      .imsi_prefix = bfromcstr(imsi_prefix);
+                }
+              }
+              if ((config_setting_lookup_string(
+                      sub2setting,
+                      MME_CONFIG_STRING_NAS_APN_CORRECTION_MAP_APN_OVERRIDE,
+                      (const char**) &apn_override))) {
+                if (config_aA->m5g_nas_config.m5g_apn_map_config.apn_map[i]
+                        .apn_override) {
+                  bassigncstr(
+                      config_aA->m5g_nas_config.m5g_apn_map_config.apn_map[i]
+                          .apn_override,
+                      apn_override);
+                } else {
+                  config_aA->m5g_nas_config.m5g_apn_map_config.apn_map[i]
+                      .apn_override = bfromcstr(apn_override);
+                }
+              }
+              config_aA->m5g_nas_config.m5g_apn_map_config.nb += 1;
+            }
+          }
+        }
+      }
+    }
+#if (!EMBEDDED_SGW)
+    // S-GW Setting
+    setting =
+        config_setting_get_member(setting_mme, MME_CONFIG_STRING_SGW_CONFIG);
+
+    if (setting != NULL) {
+      if ((config_setting_lookup_string(
+              setting, MME_CONFIG_STRING_SGW_IPV4_ADDRESS_FOR_S11,
+              (const char**) &sgw_ip_address_for_s11))) {
+        OAILOG_DEBUG(
+            LOG_MME_APP, "sgw interface IP information %s\n",
```

Comment:
is this required in AMF configuration? let us double check.

Reply:
Comments addressed.